pkg/controller/build/image_build_request.go

```diff
@@ -239,12 +225,14 @@ func (i ImageBuildRequest) toBuild() *buildv1.Build {
 				},
 				Output: buildv1.BuildOutput{
 					To: &corev1.ObjectReference{
-						Name: i.FinalImage.Pullspec,
+						Name: i.MachineOSConfig.Spec.BuildInputs.FinalImagePullspec,
```

Comment:
note: It is not obvious (and I apologize for that) that this value is mutated from what is retrieved from the `on-cluster-build-config` ConfigMap. Specifically, the name of the rendered MachineConfig replaces the user-provided tag. The user-provided tag is ignored, although maybe we should update the user-provided tag instead.

(To be clear: I'm just bringing this to your attention.)